src/nerve_tools/configure_nerve.py

```diff
@@ -282,6 +314,34 @@ def generate_subconfiguration(
             if paasta_instance:
                 config[v2_key]['labels']['paasta_instance'] = paasta_instance
 
+            if envoy_service_info:
+                envoy_key = f'{service_name}.{zk_location}:{ip_address}.{port}.envoy'
```

Comment:
This is a lot of code in an already huge function.

What if it was:

    if envoy_service_info:
        envoy_key =...
        config[envoy_key] = compose_envoy_config(...

```